index.js

```diff
@@ -1,21 +1,38 @@
 'use strict';
-module.exports = function (ms) {
-	ms = ms || 0;
-
-	var promise = new Promise(function (resolve) {
-		setTimeout(resolve, ms);
-	});
-
-	function thunk(result) {
-		return new Promise(function (resolve) {
-			setTimeout(function () {
-				resolve(result);
-			}, ms);
+module.exports = generate(true);
+module.exports.resolve = module.exports;
```

Comment:
Nice try, but we already made a decision about this in #7. We're always open to reconsider if you surface some use-cases we haven't thought of, but should be discussed in the original issue. We don't add stuff just *because*. There should always be an actual use-case behind something. It might seem simple, but everything adds overhead, either for the system or for the user.